components/relevancy/src/lib.rs

```diff
@@ -125,15 +130,12 @@ impl RelevancyStore {
     /// of success. The arm with the highest sampled probability is selected and returned.
     #[handle_error(Error)]
     pub fn bandit_select(&self, bandit: String, arms: &[String]) -> ApiResult<String> {
-        // we should cache the distribution so we don't retrieve each time
-
+        let mut cache = self.cache.lock();
```

Comment:
Can we move the `lock` inside the `for arm in arms`? This way we hold the lock of the Mutex not for the whole function. Maybe I am missing something, but it looks like we can request the `lock` in the `for` loop.

Reply:
Is this an improvement though?  There's overhead to locking the mutex multiple times and I don't think we're really giving other threads an opportunity to take the lock themselves.  Almost all of the work will be in `cache.get_beta_distribution` where we need to hold the lock.

Reply:
Yeah, that's true!